src/python/pants/engine/internals/parser.py

```diff
@@ -271,6 +271,9 @@ def __call__(self, **kwargs: Any) -> TargetAdaptor:
                 )
             kwargs["name"] = None
 
+        frame = inspect.currentframe()
+        source_line = frame.f_back.f_lineno if frame and frame.f_back else "??"
+        kwargs["description_of_origin"] = f"{self._parse_state.filepath()}:{source_line}"
```

Comment:
squatting on this field name here.. feels rather safe to assume there's not going to be a field named `description_of_origin` like ever..? 😬

Reply:
Otherwise it would make more sense to pass the field values as a single value rather than spread out with kwargs to the target adaptor ctor.

Reply:
Seems like it would be cheap to underscore-prefix it in the constructor to make collisions less likely, but not in the property name of the `Target` class? But yea, fine either way.

Reply:
yea I did that when I called it `source` for obvious reasons, not sure why I didn't just keep the underscores when changing name..